src/main/java/com/product/controller/ProductServiceController.java

```diff
@@ -1,101 +1,146 @@
+        result.put("reviews", reviewDtoList);
+        return result;
     }
 
     @PutMapping("/products/{id}")
-    public ResponseEntity<Object> updateProduct(@RequestBody Product product, @PathVariable long id) {
+    public ResponseEntity<Object> updateProduct(@RequestBody ProductDto productDto, @PathVariable long id) {
 
         Optional<Product> productOptional = productService.findProductById(id);
         if (!productOptional.isPresent())
             return ResponseEntity.notFound().build();
-        productService.getProductById(id);
 
-        product.setId(id);
+        Product product = updateProductEntity(productDto, productOptional.get());
 
         productService.addProduct(product);
-
         return ResponseEntity.noContent().build();
     }
 
 
-    @RequestMapping(value="/delete/product",method= RequestMethod.DELETE)
-    public void deleteProduct(@RequestBody String id){
-            log.info("deleting product :{}",id);
-           // productService.deleteProduct(productName);
-        productService.deleteProduct(Long.parseLong(id));
+    @DeleteMapping(value = "/product/{id}")
+    public void deleteProduct(@PathVariable long id) {
+        log.info("deleting product :{}", id);
+        productService.deleteProduct(id);
     }
 
+    @PostMapping(value = "/product/reviews/{id}")
+    public ResponseEntity<Object> addProductReview(@RequestBody ReviewDto reviewDto, @PathVariable int id) {
 
+        String uri = "http://localhost:8086/products/{productId}/reviews";
+        Map<String, String> params = new HashMap<>();
+        params.put("productId", "1");
 
-     /*@RequestMapping(value="/add/AllProducts",method = RequestMethod.POST,consumes="application/json")
-    public void addAllProducts(@RequestBody List<Product> productList){
-        if (!CollectionUtils.isEmpty(productList)){
-            log.info("storing product data:{}",productList);
-            productService.addAllProducts(productList);
+        HttpEntity<ReviewDto> request = new HttpEntity<>(reviewDto);
+        ResponseEntity<Object> responseEntity = restTemplate.exchange(uri, HttpMethod.POST, request, Object.class, params);
+        if (responseEntity.getStatusCode() != HttpStatus.CREATED) {
+            log.info("unable to add review");
         }
-    }*/
+        URI location = ServletUriComponentsBuilder.fromCurrentRequest()
+                .path("/{id}")
+                .buildAndExpand(id).toUri();
+        return ResponseEntity.created(location).build();
+    }
+
+
+    private Product updateProductEntity(ProductDto productDto, Product product) {
+
+        Product productEntity = convertToEntity(productDto);
+        productEntity.setId(product.getId());
+        if (StringUtils.isEmpty(productEntity.getName())) {
+            productEntity.setName(product.getName());
+        }
+        if (StringUtils.isEmpty(product.getDescription())) {
+            productEntity.setName(product.getDescription());
```

Comment:
Need to update properties of DB entity from dto. you are doing it opposite .please check

Reply:
I have changed the code and committed ..